dapr/clients/__init__.py

```diff
@@ -106,20 +106,70 @@ def invoke_method(
             InvokeMethodResponse: the response from the method invocation.
         """
         if self.invocation_client:
-            return self.invocation_client.invoke_method(
+            result = self.invocation_client.invoke_method(
+                app_id,
+                method_name,
+                data,
+                content_type=content_type,
+                metadata=metadata,
+                http_verb=http_verb,
+                http_querystring=http_querystring,
+                return_coroutine=False)
+        else:
+            result = super().invoke_method(
                 app_id,
                 method_name,
                 data,
                 content_type=content_type,
                 metadata=metadata,
                 http_verb=http_verb,
                 http_querystring=http_querystring)
+        if isinstance(result, InvokeMethodResponse):
+            return result
+        else:
+            raise DaprInternalError(
+                f'Failed to invoke method: Unexpected return type')
+
+    async def invoke_method_async(
+            self,
+            app_id: str,
+            method_name: str,
+            data: Union[bytes, str, GrpcMessage],
+            content_type: Optional[str] = None,
+            metadata: Optional[MetadataTuple] = None,
+            http_verb: Optional[str] = None,
+            http_querystring: Optional[MetadataTuple] = None) -> Union[InvokeMethodResponse, Awaitable[Any]]:
+        """Invoke a service method over gRPC or HTTP.
+
+        Args:
+            app_id (str): Application Id.
+            method_name (str): Method to be invoked.
+            data (bytes or str or GrpcMessage, optional): Data for requet's body.
+            content_type (str, optional): Content type of the data.
+            metadata (MetadataTuple, optional): Additional metadata or headers.
+            http_verb (str, optional): HTTP verb for the request.
+            http_querystring (MetadataTuple, optional): Query parameters.
+
+        Returns:
+            Union[InvokeMethodResponse,Awaitable[Any]]: the response from the method invocation.
+        """
+        if self.invocation_client:
+            result = self.invocation_client.invoke_method(
```

Comment:
Shouldn't be `result = await self.invocation_client.invoke_method` ? Otherwise the coroutine is never awaited.

Reply:
```suggestion
            result = await self.invocation_client.invoke_method(
```